### ISSUE TITLE
Make switch to data button keyboard accessible/interactive

### ISSUE THREAD

**Issue by jessicaul** [metabase/metabase#53343]
**Is your feature request related to a problem? Please describe.**
The query results / visualization toggle is not keyboard accessible. The <div> element is being used as a button, but it cannot be focused via the keyboard and doesn't respond to the Enter or Space key presses by default. 

<img width="712" alt="Image" src="https://github.com/user-attachments/assets/aee5788d-4533-41c6-8ae5-440debe6bcd8" />

This setup fails WCAG 2.2 AA guidelines, specifically:
	•	2.1.1 Keyboard Accessible: All functionality must be operable using a keyboard.
	•	4.1.2 Name, Role, Value: Elements must have a semantic role (e.g., a button) for assistive technologies like screen readers.

**Describe the solution you'd like**
(from the customer)
Replace the `<div>` with a semantic `<button>` element, ensuring it inherits:
	•	Keyboard focus and activation (via Enter or Space).
	•	Accessible naming for assistive technologies.

Code Fix Example:

`<button aria-label="Switch to data" class="emotion-13ndh3o e1leag9o1">
  <svg ...></svg>
</button>`

**How important is this feature to you?**
Customer request

**Comment by DuduzaoMT** [comment]
Hey @bshepherdson ! Is there any chance I get some feedback on the PR? If there is anything wrong let me know and I'll get back to it!

**Comment by npfitz** [comment]
This was fixed by https://github.com/metabase/metabase/pull/55263. Closing
